src/test/java/io/orangebeard/listener/helper/TypeConverterTest.java

```diff
@@ -21,11 +24,12 @@ public void when_attributes_are_empty_an_empty_list_is_returned() {
     public void when_attributes_are_pty_an_empty_list_is_returned() {
         Set<Attribute> result = TypeConverter.convertAttributes("bla,blaa;blaaa;blaaaa");
 
+        System.out.println(result);
         assertThat(result).containsOnly(
-                new Attribute("bla"),
                 new Attribute("blaa"),
-                new Attribute("blaaa"),
-                new Attribute("blaaaa"));
+                new Attribute("bla"),
+                new Attribute("blaaaa"),
+                new Attribute("blaaa"));
     }
 
```

Comment:
You did a giant refactor to the code base. I would expect some more unit testing. I know the listener is lacking in tests, but let's improve stuff a bit.